app/src/main/java/org/oppia/android/app/utility/DialogExtension.kt

```diff
@@ -0,0 +1,42 @@
+package org.oppia.android.app.utility
+
+import android.content.Context
+import android.content.DialogInterface
+import androidx.annotation.StyleRes
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.oppia.android.R
+
+/*
+* MaterialAlertDialogBuilder Extension
+* */
+fun Context.alertDialog(
+  @StyleRes style: Int = 0,
+  dialogBuilder: MaterialAlertDialogBuilder.() -> Unit
+) {
```

Comment:
I like where you're going with this. I think my main hesitation is that we've historically preferred using dialog fragments to host alert dialogs since it both gives us more control over the dialog's lifecycle (i.e. we can ensure that the dialog is shown synchronously via a transaction rather than asynchronously) and it provides more flexibility in centralizing the theme & style of the dialog (though I suppose we can do the same here via a central pattern).

Do you think there would be a way to generalize our alert dialogs over to a central dialog fragment, instead? I'm not sure if there's actually a potential for simplification there, but I think that's the approach we'd want to consider (and move away from alert dialogs entirely where they're only created in a single, general dialog fragment).